pkg/obfuscator/domain_test.go

```diff
@@ -114,34 +69,32 @@ func TestDomainObfuscatorContents(t *testing.T) {
 
 func TestDomainObfuscator_FileName(t *testing.T) {
 	for _, tc := range []struct {
-		name   string
-		input  string
-		output string
-		report map[string]string
+		name    string
+		input   string
+		output  string
+		domains []string
+		report  map[string]string
 	}{
 		{
-			name:   "domain with extension",
-			input:  "test.host1.log",
-			output: "test.obfuscated0001.log",
+			name:    "domain with extension",
+			domains: []string{"test.com"},
+			input:   "requests.test.com.log",
+			output:  "requests.domain0000001.log",
 			report: map[string]string{
-				"test.host1": "test.obfuscated0001",
+				"requests.test.com": "requests.domain0000001",
 			},
 		},
 		{
-			name:   "no domain with extension",
-			input:  "node1.log",
-			output: "node1.log",
-			report: map[string]string{},
-		},
-		{
-			name:   "no domain and no extension",
-			input:  "report",
-			output: "report",
-			report: map[string]string{},
+			name:    "non-matching domain",
+			domains: []string{"test.com"},
+			input:   "report.test",
+			output:  "report.test",
+			report:  map[string]string{},
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			o := NewDomainObfuscator(nil)
+			o, err := NewDomainObfuscator(tc.domains)
+			require.NoError(t, err)
 			output := o.FileName(tc.input)
 			assert.Equal(t, tc.output, output)
 			assert.Equal(t, tc.report, o.ReportingResult())
```

Comment:
please also add a test for when that regex doesn't compile it should return an error

Reply:
Done.